pkg/kubelet/apis/config/v1beta1/defaults.go

```diff
@@ -267,4 +267,7 @@ func SetDefaults_KubeletConfiguration(obj *kubeletconfigv1beta1.KubeletConfigura
 	if obj.LocalStorageCapacityIsolation == nil {
 		obj.LocalStorageCapacityIsolation = utilpointer.BoolPtr(true)
 	}
+	if obj.ImageServiceEndpoint == "" && obj.ContainerRuntimeEndpoint != "" {
+		obj.ImageServiceEndpoint = obj.ContainerRuntimeEndpoint
+	}
```

Comment:
This part of the logic is repeated many times, can it be optimized?

Reply:
Only dup once in https://github.com/kubernetes/kubernetes/pull/112136/files#diff-67dd9e8f3ee257072765326cb4f242852554a2c0753563fa51e292c0a63a7b94R289-R293.

Probably, we can remove it in `pkg/kubelet/kubelet.go` as the defaulting already handle it.

Reply:
Resolved.

Reply:
PreInitRuntimeService will not call defaulting. So this cannot be removed. I revert the change 😓.

Reply:
ok. Thanks very much.